presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -573,19 +595,24 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
                             subPlan.getRoot(),
                             entry.getValue(),
                             builder.build(),
-                            Optional.empty(),
-                            ImmutableList.of()));
+                            Optional.empty()));
         }
 
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
+                    node,
+                    "ORDER BY in aggregate function with empty grouping set");
```

Comment:
The if condition is different then the reported message. The actual message should be:
```
ORDER BY in aggregate function with at least one empty grouping set and at least one non-empty grouping set
```